### ISSUE TITLE
Bug: [use-unknown-in-catch-callback-variable] Does not flag second argument of `.then`

### ISSUE THREAD

**Issue by cobaltt7** [typescript-eslint/typescript-eslint#9056]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.tsx&code=AoJw9gtglgzgpgOhHGYA2A3OAKAlAgFwAs4A7bPAAgF4A%2BSgb0oF8AaS7OEcEXG%2Brjz4B6YZUFgQlWJQIBPAA5wAJpQCGMdaTlA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6WZRW2JgayYD2Adya1mtMgEN8ZABYTJ8eACNJZXrQBukqpOVJ0URNGgDokcGAC%2BIK0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
Promise.resolve().then(() => { }, (error) => { })
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/use-unknown-in-catch-callback-variable": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected an error as `error` is not explicitly typed as `unknown` and instead falls back to `any`.

### Actual Result

No error.

**Comment by kirkwaiblinger** [comment]
Good spot! I completely spaced on that in the initial implementation 🤣 https://github.com/typescript-eslint/typescript-eslint/pull/8383

The existing rule basically assumes catch throughout, but adding support for `.then(f, g)` should be quite straightfoward
